### PR TITLE
Refactor KubernetesIngressCreate to make testable

### PR DESCRIPTION
**Pull request by tarrow**

### CODE REVIEW

File: app/Jobs/KubernetesIngressCreate.php [hunk 1]
```diff
@@ -33,18 +34,12 @@ public static function getKubernetesIngressName( $wikiId ): string {
...
 
         // Connection example from: https://github.com/maclof/kubernetes-client#using-a-service-account
-        echo 'Creating k8s client'.PHP_EOL;
-
-        $client = new Client([
-            'master' => 'https://kubernetes.default.svc',
-            'ca_cert' => '/var/run/secrets/kubernetes.io/serviceaccount/ca.crt',
-            'token'   => '/var/run/secrets/kubernetes.io/serviceaccount/token',
-        ]);
+        Log::info( 'Creating k8s client' );
```
Comment (outdooracorn) [inline app/Jobs/KubernetesIngressCreate.php:42]: This comment should probably be moved to the [KubernetesClientServiceProvider](https://github.com/wbstack/api/pull/363/files#diff-51f8bdbbd24722c7c8dec4520ad3ca391dc4da5014817493901b831585451a51) where the `Client` is now created.

File: tests/Jobs/KubernetesIngressCreateTest.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        
+        $job->handle( new Client( [
+            [
+                'master' => 'https://kubernetes.default.svc',
+                'ca_cert' => '/var/run/secrets/kubernetes.io/serviceaccount/ca.crt',
+                'token' => '/var/run/secrets/kubernetes.io/serviceaccount/token',
+                ]
+            ], $mockGuzzle ) );
+
+    }
```
Comment (outdooracorn) [inline tests/Jobs/KubernetesIngressCreateTest.php:51]: Remove extra array and tidy up layout
```suggestion
        $job->handle(new Client([
            'master' => 'https://kubernetes.default.svc',
            'ca_cert' => '/var/run/secrets/kubernetes.io/serviceaccount/ca.crt',
            'token' => '/var/run/secrets/kubernetes.io/serviceaccount/token',
        ], $mockGuzzle));
```

File: app/Providers/KubernetesClientServiceProvider.php [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    {
+        $this->app->singleton(Client::class , function ($app) {
+            return new Client([
+            'master' => 'https://kubernetes.default.svc',
+            'ca_cert' => '/var/run/secrets/kubernetes.io/serviceaccount/ca.crt',
+            'token' => '/var/run/secrets/kubernetes.io/serviceaccount/token',
+            ]);
+        });
+    }
+}
```
Comment (outdooracorn) [inline app/Providers/KubernetesClientServiceProvider.php:25]: Nitpick: add newline at end of file
```suggestion
}

```

**Review by outdooracorn — COMMENTED** [review]
Looks good, added a few minor niggles

File: tests/Jobs/KubernetesIngressCreateTest.php [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        $handlerStack = HandlerStack::create($mock);
+        $mockGuzzle = new GuzzleHttpClient(['handler' => $handlerStack]);
+        
+        $job->handle(new Client([
+            'master' => 'https://kubernetes.default.svc',
+            'ca_cert' => '/var/run/secrets/kubernetes.io/serviceaccount/ca.crt',
+            'token' => '/var/run/secrets/kubernetes.io/serviceaccount/token',
+        ], $mockGuzzle));
+
+}
```
Comment (addshore) [inline tests/Jobs/KubernetesIngressCreateTest.php:49]: Not enough semi colons?

File: app/Jobs/KubernetesIngressCreate.php (hunk 1 above)
Comment (addshore) [reply app/Jobs/KubernetesIngressCreate.php:42]: Could just be worth removing this for now

Comment (tarrow) [inline app/Jobs/KubernetesIngressCreate.php:42]:
```suggestion
```

**Pull request merged**